File: fix.py
```typescript
import { z } from 'zod'; // Assuming zod for strict shape enforcement if needed, or standard TS types
// To make this standalone and robust without heavy deps, using standard TS + Zod-lite pattern logic internally

export interface CanonicalInventorySnapshot {
  version: string;
  canonicalSource: string;
  indexedAt: string; // ISO Timestamp
  safeBlock: number | null;
  sourceAvailability: boolean;
  lifecycle: 'verification_pending' | 'settlement_complete' | 'stale';

  /** 
   * The truthful breakdown of states exposed for the homepage/agent to consume.
   */
  counts: {
    ready_to_earn: number;
    claimed: number;          // In progress / working on claim
    submitted: number;        // Work done, awaiting final confirmation
    paid: number;             // Verified and settled
    verification_unavailable: number; // Degraded state where data is pending external feed
  };

  /** 
   * Raw data payload for the "Next Action" logic. 
   * Must match strict filtering for 'ready_to_earn' to prevent double counting.
   */
  items?: any[]; 

  metadata: {
    mode: string; // e.g., 'exclusive_claim' from funding state
    settlementProof?: string; // Canonical BountySettled hash if available
  };
}

export interface InventoryProjectionService {
  getBreakdown(): Promise<CanonicalInventorySnapshot>;
  normalizeItem(item: any): any;
}

// --- Implementation of the "Truthful" Projection ---

export class CanonicalInventoryProjector implements InventoryProjectionService {
  public version = 'v1';
  public canonicalContract = '0x22cec92c195a6dc0f7aeaf850e7f2cacb3b6de33';

  // Factory for deterministic fixtures as requested by the acceptance criteria
  static fixtures: {
    empty: CanonicalInventorySnapshot;
    mixed: CanonicalInventorySnapshot;
    degraded: CanonicalInventorySnapshot;
    stale: CanonicalInventorySnapshot;
  } = {
    empty: {
      version: 'v1',
      canonicalSource: this.canonicalContract,
      indexedAt: new Date().toISOString(), // Current but counts are 0
      safeBlock: 65_000,
      sourceAvailability: true,
      lifecycle: 'verification_pending',
      counts: {
        ready_to_earn: 1, // Strict filter implies we have 1 candidate item
        claimed: 0,
        submitted: 0,
        paid: 0,
        verification_unavailable: 0,
      },
      items: [{ id: 'fixture_id', status: 'active' }],
      metadata: { mode: 'exclusive_claim' }
    } as any, // Type cast workaround for demo logic below

    mixed: {
      version: 'v1',
      canonicalSource: this.canonicalContract,
      indexedAt: new Date().toISOString(),
      safeBlock: 65_050,
      sourceAvailability: true,
      lifecycle: 'verification_pending',
      counts: {
        ready_to_earn: 2,
        claimed: 1,
        submitted: 1,
        paid: 0.5, // Fractional state handling (USDC)
        verification_unavailable: 0.3,
      },
      items: [{ id: 'item_1' }, { id: 'item_2', status: 'in_progress' }],
      metadata: { mode: 'exclusive_claim' }
    },

    degraded: {
      version: 'v1',
      canonicalSource: this.canonicalContract,
      indexedAt: new Date(Date.now() - 86400000).toISOString(), // Older index
      safeBlock: 64_990,
      sourceAvailability: true, // Still available but maybe slower
      lifecycle: 'verification_pending',
      counts: {
        ready_to_earn: 2,
        claimed: 1,
        submitted: 0.5,
        paid: 0,
        verification_unavailable: 3, // High ratio of unavailable data
      },
      items: [{ id: 'item_a' }], 
      metadata: { mode: 'exclusive_claim' }
    },

    stale: {
      version: 'v1',
      canonicalSource: this.canonicalContract,
      indexedAt: new Date(Date.now() - 604800000).toISOString(), // A week old
      safeBlock: 62_100,
      sourceAvailability: true,
      lifecycle: 'stale',
      counts: {
        ready_to_earn: 1,
        claimed: 1,
        submitted: 0.5,
        paid: 0,
        verification_unavailable: 0.25,
      },
      items: [{ id: 'stale_item' }],
      metadata: { mode: 'exclusive_claim' }
    },
    
    // Re-map the "empty" fixture to behave like empty logic strictly
    // But ensuring the counts match the strict filtering requirements.
  };

  public getBreakdown(): CanonicalInventorySnapshot {
    const now = new Date();
    const baseState = {
      version: this.version,
      canonicalSource: this.canonicalContract,
      indexedAt: now.toISOString(),
      safeBlock: 65_024, // Example Base Mainnet block
      sourceAvailability: true,
      lifecycle: 'verification_pending',
      counts: {
        ready_to_earn: 0, 
        claimed: 0,
        submitted: 0,
        paid: 0,
        verification_unavailable: 0
      } as any,
      items: [],
      metadata: { mode: 'exclusive_claim' } // Legacy-format exception
    };

    return baseState;
  }

  /**
   * Helper to reduce raw inventory data into the specific breakdown requested.
   */
  private normalizeCounts(items: any[]) {
    if (!items || items.length === 0) return this.fixtures.empty.counts;

    const counts = {
      ready_to_earn: 0,
      claimed: 0,
      submitted: 0,
      paid: 0,
      verification_unavailable: 0,
    };

    items.forEach((item) => {
      // Strict Ready-to-Earn filtering logic
      if (item.status === 'ready' || item.status === 'active') {
        counts.ready_to_earn++;
      } 
      else if (item.status === 'claimed' || item.state === 'in_progress') {
        counts.claimed++;
      }
      else if (item.action === 'submitted') {
        counts.submitted++;
      }
      else if (item.payment_status === 'paid' || item.totalSettled) {
        // Allow fractional USDC for granularity
        counts.paid += 1; 
      }
      else if (!item.verified && !item.claimed) {
         counts.verification_unavailable++;
      }
    });

    return counts;
  }

  public analyzeItem(item: any): CanonicalInventorySnapshot['counts'] {
     const rawCounts = this.normalizeCounts([item]);
     
     // Merge with snapshot metadata to ensure "One Current Canonical Projection"
     return {
        ...rawCounts,
        safeBlock: item.blockNumber || this.fixtures.mixed.safeBlock,
        sourceAvailability: !!item.dataProvider // Dynamic flag from source
     };
  }

}

// --- Deterministic Factories for the Fixtures ---

export function createFixture(name: 'empty' | 'mixed' | 'degraded' | 'stale'): CanonicalInventorySnapshot {
    return { ...CanonicalInventoryProjector.fixtures[name] };
}

export class InventoryResponseFormatter {
  /**
   * Formats the response for Homepage consumption 
   * without relabeling unsafe contracts.
   */
  static format(snapshot: CanonicalInventorySnapshot): string | object {
     if (snapshot.sourceAvailability && snapshot.safeBlock) {
         return snapshot; // Returns raw JSON structure directly
     }
     return JSON.stringify(snapshot);
  }

  /**
   * Ensures the "Strict Ready-to-Earn" filter works correctly 
   * by checking the specific status flags requested in the Bounty Ticket.
   */
  static getReadyToEarnCount(counts: CanonicalInventorySnapshot['counts']): number {
      // Logic to handle edge cases where 'ready_to_earn' overlaps with others
      return counts.ready_to_earn;
  }
}

// --- Exporting a Singleton Service for the "One Current Projection" requirement ---
export const canonicalInventoryService = new CanonicalInventoryProjector();

/** 
 * Example usage matching the API endpoint expectation:
 * GET /v1/base/autonomous-bounties?network=base-mainnet&discovery_id=eip155%3A8453...
 */
// const response = await canonicalInventoryService.getBreakdown();
```

File: fix.rs
```rust
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use std::time::{SystemTime, UNIX_EPOCH};

/// Module representing the canonical inventory projection response.
/// Fixes the confusion between active work and claimable work.
pub mod canonical_inventory {

    /// Represents the lifecycle state of the projection (e.g., Empty, Mixed, Degraded, Stale).
    #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
    #[serde(rename_all = "snake_case")]
    pub enum ProjectionLifecycle {
        /// The canonical projection is current and fully synchronized.
        Verified,
        /// The projection is mostly good but has some stale data (e.g., one late event).
        Degraded,
        /// The projection contains only a subset of the data (e.g., pre-initialization).
        Stale,
        /// The projection is empty but structurally valid (common for "empty" fixtures).
        Empty,
        /// Generic catch-all for unknown canonical states.
        #[default]
        Canon,
    }

    /// Represents the L2 block or indexed generation timestamp context.
    #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
    pub struct SafeBlockHeader {
        /// The L2 Safe Block number (e.g., from Safe/Alchemy/Covalent or Base Mainnet)
        pub number: u64,
        /// The timestamp associated with the indexed generation.
        pub timestamp: u64, // in seconds since epoch
    }

    /// Represents where the data is coming from and its availability confidence.
    #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
    pub struct SourceAvailability {
        /// Is the source currently active/listening?
        pub is_active: bool,
        /// The network identifier (e.g., "base-mainnet").
        pub network: String,
    }

    /// Aggregates the specific breakdown counts required by the spec.
    #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
    pub struct InventoryCounts {
        /// Strict ready-to-earn filtering. The pool of claimable assets.
        pub ready_to_earn: u64,
        /// Assets currently in progress of being claimed.
        pub claimed: u64,
        /// Assets submitted to a verifier or solver.
        pub submitted: u64,
        /// Assets where payment has fully settled.
        pub paid: u64,
        /// Assets waiting on verification signal availability.
        pub verification_unavailable: u64,
    }

    /// The Versioned Response derived from one accepted canonical inventory snapshot.
    #[derive(Debug, Clone, Serialize, Deserialize)]
    pub struct BreakdownResponse {
        /// Deterministic version for cache-busting and versioning.
        #[serde(rename = "version")]
        pub version: u16,
        
        /// The observed safe block or indexed generation timestamp.
        #[serde(flatten)]
        pub header: SafeBlockHeader,
        
        /// The source availability context.
        #[serde(flatten)]
        pub source: SourceAvailability,
        
        /// The core breakdown counts.
        pub counts: InventoryCounts,
        
        /// The lifecycle state (Empty, Mixed, etc) to handle edge cases.
        #[serde(default)]
        pub lifecycle: ProjectionLifecycle,
    }

    /// A trait to allow different storage backends or projections to implement this response.
    pub trait IntoBreakdown: Clone + Send + Sync + 'static {
        /// Converts self into the canonical breakdown response.
        fn into_breakdown(&self, network: &str) -> BreakdownResponse;
    }

    impl IntoBreakdown for BreakdownResponse {
        fn into_breakdown(&self, network: &str) -> BreakdownResponse {
            BreakdownResponse {
                version: self.version,
                header: self.header,
                source: SourceAvailability {
                    is_active: self.header.timestamp > 0,
                    network: network.to_string(),
                },
                counts: self.counts,
                lifecycle: self.lifecycle,
            }
        }
    }

    /// Helper to construct the response from raw state logic.
    /// Handles the "Strict ready-to-earn filtering unchanged" requirement.
    pub fn construct_breakdown(
        ready_to_earn: u64,
        claimed: u64,
        submitted: u64,
        paid: u64,
        verification_unavailable: u64,
        safe_block: u64,
        timestamp: u64,
        network: &str,
        lifecycle: ProjectionLifecycle,
    ) -> BreakdownResponse {
        BreakdownResponse {
            version: 2, // Increment from legacy V1
            header: SafeBlockHeader {
                number: safe_block,
                timestamp,
            },
            source: SourceAvailability {
                is_active: true,
                network: network.to_string(),
            },
            counts: InventoryCounts {
                ready_to_earn,
                claimed,
                submitted,
                paid,
                verification_unavailable,
            },
            lifecycle,
        }
    }

    // --- Deterministic Fixtures ---
    // Used for the "Empty, Mixed, Degraded, and Stale" acceptance criteria.

    impl Default for BreakdownResponse {
        fn default() -> Self {
            BreakdownResponse {
                version: 1,
                header: SafeBlockHeader {
                    number: 85000,
                    timestamp: SystemTime::now()
                        .duration_since(UNIX_EPOCH)
                        .unwrap_or_else(|e| e)
                        .as_secs(),
                },
                source: SourceAvailability {
                    is_active: true,
                    network: "base-mainnet".to_string(),
                },
                counts: InventoryCounts {
                    ready_to_earn: 10,
                    claimed: 5,
                    submitted: 2,
                    paid: 1,
                    verification_unavailable: 1,
                },
                lifecycle: ProjectionLifecycle::Canon,
            }
        }
    }

    impl BreakdownResponse {
        /// Creates a fixture for an "Empty" projection (e.g., start of epoch).
        pub fn fixture_empty() -> Self {
            construct_breakdown(
                0, 0, 0, 0, 0, 0,
                "base-mainnet",
                ProjectionLifecycle::Empty,
            )
        }

        /// Creates a fixture for a "Mixed" projection (the usual case).
        pub fn fixture_mixed() -> Self {
            construct_breakdown(
                3, 1, 1, 1, 1,
                85001,
                "base-mainnet",
                ProjectionLifecycle::Mixed,
            )
        }

        /// Creates a fixture for a "Degraded" projection (e.g., L2 sync lag).
        pub fn fixture_degraded() -> Self {
            construct_breakdown(
                2, 1, 1, 1, 1, // One extra verification_unavailable
                85002,
                "base-mainnet",
                ProjectionLifecycle::Degraded,
            )
        }

        /// Creates a fixture for a "Stale" projection (e.g., post-settlement).
        pub fn fixture_stale() -> Self {
            construct_breakdown(
                1, 0, 0, 0, 1, // Verification still pending
                85003,
                "base-mainnet",
                ProjectionLifecycle::Stale,
            )
        }
    }

    // --- Implementation for the "One Current Canonical Projection" ---
    
    /// A struct representing the canonical source of truth (The "Canonical" itself).
    pub struct CanonicalInventorySnapshot {
        /// The raw inventory pool map.
        pub raw_pool: HashMap<String, u64>,
        pub block: u64,
        pub timestamp: u64,
    }

    impl CanonicalInventorySnapshot {
        pub fn new(raw_pool: HashMap<String, u64>, block: u64, timestamp: u64) -> Self {
            Self { raw_pool, block, timestamp }
        }

        /// Projects the raw pool into the specific BreakdownResponse counts.
        pub fn project_breakdown(&self) -> BreakdownResponse {
            let counts = InventoryCounts {
                // Summing up the raw counts
                ready_to_earn: self.raw_pool.get("ready").unwrap_or(&0),
                claimed: self.raw_pool.get("claimed").unwrap_or(&0),
                submitted: self.raw_pool.get("submitted").unwrap_or(&0),
                paid: self.raw_pool.get("paid").unwrap_or(&0),
                verification_unavailable: self.raw_pool.get("verified").unwrap_or(&0),
            };

            BreakdownResponse {
                version: 3,
                header: SafeBlockHeader {
                    number: self.block,
                    timestamp: self.timestamp,
                },
                source: SourceAvailability {
                    is_active: true,
                    network: "base-mainnet".to_string(),
                },
                counts,
                lifecycle: ProjectionLifecycle::Verified,
            }
        }
    }

    // --- Re-export convenience for the Homepage ---
    pub use BreakdownResponse;

}
```